pkg/rkenodeconfigclient/certificates_test.go

```diff
@@ -8,13 +8,107 @@ import (
 	"crypto/x509"
 	"crypto/x509/pkix"
 	"encoding/pem"
-	"github.com/stretchr/testify/assert"
 	"math/big"
 	"net"
 	"testing"
 	"time"
+
+	"github.com/rancher/rke/types"
+	"github.com/stretchr/testify/assert"
 )
 
+func TestFindCommandValue(t *testing.T) {
+	commands := []string{
+		"--hostname=test.com",
+		"--tls-cert-file=a-test-value",
+		"---abc=123",
+	}
+
+	type testcase struct {
+		testName string
+		flag     string
+		value    string
+	}
+
+	tests := []testcase{
+		{
+			testName: "successfully find flag in command arguments",
+			value:    "a-test-value",
+			flag:     "--tls-cert-file",
+		},
+		{
+			testName: "unsuccessfully find flag in command arguments",
+			value:    "",
+			flag:     "--not-a-flag",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.testName, func(t *testing.T) {
+			v := findCommandValue(tc.flag, commands)
+			if v != tc.value {
+				t.Logf("wanted %s, got %s", tc.value, v)
```

Comment:
I'd generally use `assert.Equal` for this but I'm not going to hold up your PR any longer.